src/deploy/kubernetes-dashboard-arm-head.yaml

```diff
@@ -16,54 +16,64 @@
 #
 # Example usage: kubectl create -f <this_file>
 
-kind: List
-apiVersion: v1
-items:
-- kind: ReplicationController
-  apiVersion: v1
-  metadata:
-    labels:
-      app: kubernetes-dashboard-head
-    name: kubernetes-dashboard-head
-    namespace: kube-system
-  spec:
-    replicas: 1
-    selector:
-      app: kubernetes-dashboard-head
-    template:
-      metadata:
-        labels:
-          app: kubernetes-dashboard-head
-      spec:
-        containers:
-        - name: kubernetes-dashboard-head
-          image: kubernetesdashboarddev/kubernetes-dashboard-arm:head
-          imagePullPolicy: Always
-          ports:
-          - containerPort: 9090
-            protocol: TCP
-          args:
-            # Uncomment the following line to manually specify Kubernetes API server Host
-            # If not specified, Dashboard will attempt to auto discover the API server and connect
-            # to it. Uncomment only if the default does not work.
-            # - --apiserver-host=http://my-address:port
-          livenessProbe:
-            httpGet:
-              path: /
-              port: 9090
-            initialDelaySeconds: 30
-            timeoutSeconds: 30
-- kind: Service
-  apiVersion: v1
-  metadata:
-    labels:
-      app: kubernetes-dashboard-head
-    name: dashboard-head
-    namespace: kube-system
-  spec:
-    type: NodePort
-    ports:
-    - port: 80
-      targetPort: 9090
-    selector:
+kind: Deployment
+apiVersion: extensions/v1beta1
+metadata:
+  labels:
+    app: kubernetes-dashboard-head
+  name: kubernetes-dashboard-head
+  namespace: kube-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
       app: kubernetes-dashboard-head
+  template:
+    metadata:
+      labels:
+        app: kubernetes-dashboard-head
+      # Comment the following annotation if Dashboard must not be deployed on master
+      annotations:
```

Comment:
Maybe we should use tolerations in the same way as you have introduced. This might not be up to date. It's also better to keep one style across files so people won't get confused.

```yaml
tolerations:
    - key: node-role.kubernetes.io/master
      effect: NoSchedule
```